### DISCUSSION TITLE
Why does it show the wrong color scheme if I cd to ./public and start a web server?

### DISCUSSION THREAD

**Discussion by garyrob** [hexojs/hexo#5551] [Question / Answer]
I am experimenting with hexo (using the cactus theme).

I have colorscheme set to white in _config.yml (and I get the same results whether I use hexo's _config.yml or that of the cactus theme).

When I run `hexo generate` and then `hexo server`, my site shows up  in my browser as white. But when I cd to the public directory and run `python -m http.server 8000` or `caddy file-server`, and open the site, it shows up what appears to be the dark colorscheme.  But everything else about it seems to work perfectly, include all other styling.

I can't understand why that would happen. I want to be able to deploy the files to my own server and run caddy in the directory that contains them. But I need to be able to control the colorscheme. Given that your deployment docs say, "All generated files are saved in the public folder. You can copy them to wherever you like," it seems like I should be able to run the site from the public directory just as I would if I copied those files to another server.

Any thoughts??

**Comment by garyrob** [comment]
I guess I could run a reverse proxy with caddy in front of `hexo server -s` to get https; I assume such a setup could be secure for production?

**Reply by uiolee** [reply]
`hexo server` just for preview not production

**Comment by uiolee** [comment]
> colorscheme set to white in _config.yml

https://hexo.io/docs/configuration.html

as i know, colorshceme is not a config item of hexo. It may be related to theme

**Comment by uiolee** [comment]
I can't reproduce the problem you mentioned. In my tests, cactus' colorscheme works fine with any server

**Comment by garyrob** [comment]
> as i know, colorshceme is not a config item of hexo. It may be related to theme

As I understand it, there's a `theme_config` extension in hexo's _config.yml in which configuration settings in the theme can be overridden. However, whether I use that or directly change Cactus' _config.yml's colorscheme does not appear to affect the results.

> I can't reproduce the problem you mentioned. In my tests, cactus' colorscheme works fine with any server

This is extremely perplexing. Everything about my site works fine with the dev server, and also works fine with another server like Caddy, EXCEPT that with the other server, the colorscheme is dark!!  It's quite frustrating and may mean I can't use hexo/cactus. 

I `cd` to public and run  Caddy. Or `python -m http.server`. Same results: dark mode regardless of the colorscheme setting. Everything else working as it should. Changes to, for example, `nav` take effect as expected. It's only that one setting that is ignored. Is there anything you can think of that I should be doing differently that just running the server with public as the current directory?

> hexo server just for preview not production

I know, but my understanding is that it would be safe for a low-volume site if behind a real web server like nginx doing reverse proxy. Is that wrong?

**Comment by garyrob** [comment]
I just deployed it to Netlify. Same thing! Light in `hexo serve`, dark mode in netlify. Very frustrating.

**Comment by garyrob** [comment]
I could upload my site somewhere if that would be of interest. It's not secret. There must be some reason this is happening, and if it's happening to me, it seems like it could be happening to others.

**Reply by stevenjoezhang** [reply]
If the problem still persists, you can upload the source code of your blog to a repository (removing any personal information or sensitive data), and then let others reproduce and debug the issue.
